src/point_one/fusion_engine/messages/device.h

```diff
@@ -0,0 +1,44 @@
+/**************************************************************************/ /**
+ * @brief Device-specific messages.
+ * @file
+ ******************************************************************************/
+
+#pragma once
+
+#include "point_one/fusion_engine/common/portability.h"
+#include "point_one/fusion_engine/messages/defs.h"
+
+namespace point_one {
+namespace fusion_engine {
+namespace messages {
+
+// Enforce 4-byte alignment and packing of all data structures and values.
+// Floating point values are aligned on platforms that require it. This is done
+// with a combination of setting struct attributes, and manual alignment
+// within the definitions. See the "Message Packing" section of the README.
+#pragma pack(push, 1)
+
+struct alignas(4) SystemStatusMessage : public MessagePayload {
+  static constexpr MessageType MESSAGE_TYPE = MessageType::SYSTEM_STATUS;
+  static constexpr uint8_t MESSAGE_VERSION = 0;
+  static constexpr int16_t INVALID_TEMPERATURE = INT16_MIN;
+
+  /** The time of the message, in P1 time (beginning at power-on). */
+  Timestamp p1_time;
+
+  /**
+   * The temperature of the GNSS receiver.
+   *
+   * Stored in units of 1/8 degrees Celsius: `gnss_temperature_degc =
+   * gnss_temperature * 1/8`. Set to `-32768` if invalid.
+   */
+  int16_t gnss_temperature = INVALID_TEMPERATURE;
+
+  uint8_t reserved[131] = {0};
```

Comment:
Where does 131 come from?

This should probably be 118: 128 B - 2 (`int16_t`) - 8 (`Timestamp`)

Reply:
A couple things here: 
1. I thought we wanted 128 _reserved_ bytes, not 128B for the total size
2. So you don't include the `static constexpr` values in this calculation? I got a value of `131` by including those values in the calculation, and then setting it up to where we had >= 128 reserved bytes

Reply:
1. @axlan what do you think?
   - @wbrannon to be clear though, the message size must be a multiple of 4B so if we're going for ~128B reserved it either needs to be 126 or 130 so we account for the extra 2 we need since `gnss_temperature` is only 2B.
2. No, the `static` means it's a class-level variable -- a compile time constant in this case since it's `constexpr`. It's not part of an individual instance of the class, so it doesn't count towards the memory used by a message.